apps/settings/js/panels/achievements/achievements_list.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * UI for Achievements panel's functionality.
+ */
+define(function(require) {
+  'use strict';
+
+  var SettingsService = require('modules/settings_service');
+  var ListView = require('modules/mvvm/list_view');
+  var ObservableArray = require('modules/mvvm/observable_array');
+  var AchievementItemTemplateFactory = require(
+    'panels/achievements/achievements_template');
+
+  function AchievementsList(panel) {
+    this.listElement = panel.querySelector('.achievements-list');
+    this.noAchievementsElement = panel.querySelector('.no-achievements');
+
+    var achievementTemplate = AchievementItemTemplateFactory(achievement => {
+      SettingsService.navigate('achievement-details',
+        { achievement: achievement });
+    });
+
+    this._listView = ListView(this.listElement, this.achievements,
+      achievementTemplate);
+
+    this.onLengthUpdated = this.onLengthUpdated.bind(this);
+
+    this.onLengthUpdated();
+    this.achievements.observe('length', this.onLengthUpdated);
+  }
+
+  AchievementsList.prototype = {
+    get achievements() {
```

Comment:
This seems worrying because the getter of `achievements` returns an `ObservableArray` but the setter accepts an plain object.